app/models/vm_scan.rb

```diff
@@ -600,6 +556,32 @@ def abort_retry(*args)
 
   private
 
+  def create_snapshot(vm)
```

Comment:
Personally, I'd prefer all of the returns to have an explicit value, I think it makes it easier to read.
However, the implied return at the end makes the code more fragile (must be Italian); if someone adds a log message or something at the end, the code will break.